pkg/conn/basedb.go

```diff
@@ -66,27 +66,23 @@ func (d *defaultDBProvider) Apply(config config.DBConfig) (*BaseDB, error) {
 
 // BaseDB wraps *sql.DB
 type BaseDB struct {
+	sync.Mutex
```

Comment:
Is `mu sync.Mutex` better, then the outer caller can not do some call like `baseDB.Lock`?

Reply:
oh I got it, I'll update later